File: src/main/java/com/example/day03/leetcode_707/MyLinkedList.py
class MyLinkedList:

    def __init__(self):
        self.head = ListNode(0)
        self.size = 0
    def get(self, index: int) -> int:
        if index < 0 or index >= self.size:
            return -1
        cur = self.head
        for i in range(index + 1):
            cur = cur.next
        return cur.val
    def addAtHead(self, val: int) -> None:
        self.addAtIndex(0, val)
    def addAtTail(self, val: int) -> None:
        self.addAtIndex(self.size, val)

    def addAtIndex(self, index: int, val: int) -> None:
        if index > self.size:
            return
        index = max(0,index)
        self.size += 1
        pre = self.head
        for _ in range(index):
            pre = pre.next
        toAdd = ListNode(val)
        toAdd.next = pre.next
        pre.next = toAdd
    def deleteAtIndex(self, index: int) -> None:
        if index < 0 or index >= self.size:
            return
        self.size -= 1
        pre = self.head
        for _ in range(index):
            pre = pre.next
        pre.next = pre.next.next
class ListNode:
    def __init__(self,val = 0,next = None):
        self.val = val
        self.next = None

File: src/main/java/com/example/day03/leetcode_707/MyLinkedList.java
package com.example.day03.leetcode_707;


import com.example.day03.leetcode_203.ListNode;

class MyLinkedList {
    int size;
    ListNode1 head;
    public MyLinkedList() {
        size = 0;
        head = new ListNode1(0);
    }
    
    public int get(int index) {
      if(index < 0 || index >= size){
          return -1;
      }
      ListNode1 cur = head;
      for(int i = 0;i < index + 1;i ++){
          cur = cur.next;
      }
      return cur.val;
    }
    
    public void addAtHead(int val) {
        addAtIndex(0,val);
    }
    
    public void addAtTail(int val) {
        addAtIndex(size,val);
    }
    
    public void addAtIndex(int index, int val) {
        if(index > size){
            return;
        }
        index = Math.max(0,index);
        size++;
        ListNode1 pre = head;
        for(int i = 0;i < index;i ++){
            pre = pre.next;
        }
        ListNode1 toAdd = new ListNode1(val);
        toAdd.next = pre.next;
        pre.next = toAdd;
    }
    
    public void deleteAtIndex(int index) {
        if(index < 0 || index >= size){
            return;
        }
        size--;
        ListNode1 pre = head;
        for(int i = 0;i < index;i ++){
            pre = pre.next;
        }
        pre.next = pre.next.next;
    }
}

/**
 * Your MyLinkedList object will be instantiated and called as such:
 * MyLinkedList obj = new MyLinkedList();
 * int param_1 = obj.get(index);
 * obj.addAtHead(val);
 * obj.addAtTail(val);
 * obj.addAtIndex(index,val);
 * obj.deleteAtIndex(index);
 */
class ListNode1{
    int val;
    ListNode1 next;
    public ListNode1(int val) {
        this.val = val;
        this.next = null;
    }
}
